tests/integrations/github/cicd/test_github_commands.py

```diff
@@ -465,6 +466,144 @@ def test_run_all_test_failed(
+        == """**Num Successful Tests: 0**
+
+
+```some error```
+
+
+"""
+    )
+
+    assert "SQLMesh - Prod Plan Preview" in controller._check_run_mapping
+    prod_plan_preview_checks_runs = controller._check_run_mapping[
+        "SQLMesh - Prod Plan Preview"
+    ].all_kwargs
+    assert len(prod_plan_preview_checks_runs) == 2
+    assert GithubCheckStatus(prod_plan_preview_checks_runs[0]["status"]).is_queued
+    assert GithubCheckStatus(prod_plan_preview_checks_runs[1]["status"]).is_completed
+    assert GithubCheckConclusion(prod_plan_preview_checks_runs[1]["conclusion"]).is_skipped
+    assert (
+        prod_plan_preview_checks_runs[1]["output"]["title"]
+        == "Skipped generating prod plan preview since PR was not synchronized"
+    )
+    assert (
+        prod_plan_preview_checks_runs[1]["output"]["summary"]
+        == "Unit Test(s) Failed so skipping creating prod plan"
+    )
+
+    assert "SQLMesh - PR Environment Synced" in controller._check_run_mapping
+    pr_checks_runs = controller._check_run_mapping["SQLMesh - PR Environment Synced"].all_kwargs
+    assert len(pr_checks_runs) == 2
+    assert GithubCheckStatus(pr_checks_runs[0]["status"]).is_queued
+    assert GithubCheckStatus(pr_checks_runs[1]["status"]).is_completed
+    assert GithubCheckConclusion(pr_checks_runs[1]["conclusion"]).is_skipped
+    assert pr_checks_runs[1]["output"]["title"] == "PR Virtual Data Environment: hello_world_2"
+
+    assert "SQLMesh - Prod Environment Synced" in controller._check_run_mapping
+    prod_checks_runs = controller._check_run_mapping["SQLMesh - Prod Environment Synced"].all_kwargs
+    assert len(prod_checks_runs) == 2
+    assert GithubCheckStatus(prod_checks_runs[0]["status"]).is_queued
+    assert GithubCheckStatus(prod_checks_runs[1]["status"]).is_completed
+    assert GithubCheckConclusion(prod_checks_runs[1]["conclusion"]).is_skipped
+
+    assert "SQLMesh - Has Required Approval" in controller._check_run_mapping
+    approval_checks_runs = controller._check_run_mapping[
+        "SQLMesh - Has Required Approval"
+    ].all_kwargs
+    assert len(approval_checks_runs) == 3
+    assert GithubCheckStatus(approval_checks_runs[0]["status"]).is_queued
+    assert GithubCheckStatus(approval_checks_runs[1]["status"]).is_in_progress
+    assert GithubCheckStatus(approval_checks_runs[2]["status"]).is_completed
+    assert GithubCheckConclusion(approval_checks_runs[2]["conclusion"]).is_success
+
+    assert len(controller._context.apply.call_args_list) == 0
+
+    assert not mock_pull_request.merge.called
+    assert not controller._context.invalidate_environment.called
+
+    assert len(created_comments) == 0
+
+    with open(github_output_file, "r") as f:
+        output = f.read()
+        assert (
+            output
+            == "run_unit_tests=failure\nhas_required_approval=success\npr_environment_name=hello_world_2\npr_environment_synced=skipped\nprod_plan_preview=skipped\nprod_environment_synced=skipped\n"
+        )
```

Comment:
I don't know why Github represents this as an add but all of this logic already exists and is unchanged. Only lines 469 - 479 were added.